modin/pandas/groupby.py

```diff
@@ -904,19 +905,35 @@ def aggregate(self, func=None, *args, engine=None, engine_kwargs=None, **kwargs)
 
         do_relabel = None
         if isinstance(func, dict) or func is None:
-            relabeling_required, func_dict, new_columns, order = reconstruct_func(
+            # the order from `reconstruct_func` cannot be used correctly if there
+            # is more than one columnar partition
+            old_kwargs = dict(kwargs)
+            relabeling_required, func_dict, new_columns, _ = reconstruct_func(
                 func, **kwargs
             )
 
             if relabeling_required:
 
                 def do_relabel(obj_to_relabel):  # noqa: F811
-                    new_order, new_columns_idx = order, pandas.Index(new_columns)
+                    # unwrap nested labels into one level tuple
+                    labels = [None] * len(old_kwargs)
+                    for idx, x in enumerate(old_kwargs.values()):
+                        if is_scalar(x) or callable(x):
+                            labels[idx] = x if not callable(x) else x.__name__
+                            continue
+                        new_elem = []
+                        for y in x:
```

Comment:
Let's give x and y more descriptive names.

Reply:
renamed